### DISCUSSION TITLE
pacs00800110 parse SplmtryData failed

### DISCUSSION THREAD

**Discussion by daidai5233** [prowide/prowide-iso20022#95] [Q&A]
I want to parse EnvIp Content with any node inside，aspect to get an Element object，but get Error below:

com.prowidesoftware.swift.model.mx.NamespaceAndElementFilter startElement
WARNING: Error parsing Cnts [urn:iso:std:iso:20022:tech:xsd:pacs.008.001.10] element
java.lang.NegativeArraySizeException
...

SplmtryData content like below:

```
<SplmtryData>
    <PlcAndNm>XXX1</PlcAndNm>
    <Envlp>
        <Cnts>
            <Data>XXX2</Data>
        </Cnts>
    </Envlp>
</SplmtryData>
```

**Comment by daidai5233** [comment]
has found the bug:
NamespaceAndElementFilter  need  override endPrefixMapping to skip this bugs

**Comment by ptorres-prowide** [comment]
Fixed. The NegativeArraySizeException came from NamespaceAndElementFilter forwarding every endPrefixMapping, which underflowed the downstream JAXB namespace stack. As you correctly diagnosed, the fix was to stop forwarding them — endPrefixMapping is now a no-op and the matching end-mappings are synthesized in a balanced way. The no-namespace SplmtryData/Envlp/Cnts/Data payload now parses and is captured as a DOM Element. Shipped in the 10.x line; regression covered by WildcardAnyParsingTest. Thanks for the report and the spot-on root cause.
